In [2]:
from scipy.optimize import linprog

# OBJECTIVE VUNC 
# 3a + 5b + 4c = P
c = [-3, -5, -4]  

# CONSTRAINTS
# 2a + 3b <= 8
# 3a + 2b + 2c <= 15
# 4b + 5c <= 10
A = [[2, 3, 0],  
    [3, 2, 2],  
    [0, 4, 5]]

b = [8, 15, 10]  


result = linprog(c, A_ub=A, b_ub=b, bounds=(0, None))


print(f"Optimal value of P: {-result.fun}")
print(f"Optimal values of a, b, c: {result.x}")



Optimal value of P: 19.34146341463415
Optimal values of a, b, c: [3.63414634 0.24390244 1.80487805]


The maximum profit the company could expect is approximately Rs. 19.34. Rounding the optimal values of a, b, and c, they could maximize their profit if they process about 4 units of product A and 2 units of product C.